python/pyarrow/includes/libarrow_parquet_readwrite.pxd

```diff
@@ -0,0 +1,32 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# distutils: language = c++
+
+from pyarrow.includes.libarrow_dataset cimport *
+from pyarrow._parquet cimport *
+
+cdef extern from "arrow/dataset/api.h" namespace "arrow::dataset" nogil:
```

Comment:
Hmm... I'm not sure what the point is of moving these declarations to a new file?

Reply:
I assume this is related to having the separate pxd file for encryption (see comment below https://github.com/apache/arrow/pull/34616/files#r1317199494)? 
If we don't split those in a separate file, it might not work to do the

```
from pyarrow.includes.libarrow_dataset_parquet cimport *

IF PARQUET_ENCRYPTION_ENABLED:
    from pyarrow.includes.libarrow_parquet_readwrite_encryption cimport *
    from pyarrow._parquet_encryption cimport *
ELSE:
    from pyarrow.includes.libarrow_parquet_readwrite cimport *
```

Because otherwise if `CParquetFileWriteOptions` was also defined in `libarrow_dataset_parquet.pxd`, importing it from `libarrow_parquet_readwrite_encryption` would duplicate the imported declaration (and not sure if cython would be happy about that)